zip.go

```diff
@@ -186,6 +186,15 @@ func (z *Zip) extractNext(to string) error {
 	if !ok {
 		return fmt.Errorf("expected header to be zip.FileHeader but was %T", f.Header)
 	}
+	if (header.FileInfo().Mode() & os.ModeSymlink) != 0 {
+		buffer := make([]byte, header.FileInfo().Size())
```

Comment:
This seems to be very different from how it works in tar archives; while I know that this PR writes the target filepath as the symlink's contents, is it safe to assume that every zip archiver will do that? Is there risk here of reading in a potentially huge file?

Reply:
I'm not 100% sure; I simply went by the symlink code in archive/zip test: https://github.com/golang/go/blob/b0a53d2/src/archive/zip/writer_test.go#L54-L59

I believe this is what Info-ZIP refers to as the "generic symlink support" in the later versions. Earlier versions (and PKZIP) had platform specific extensions for symlinks (so symlinks archived on one platform could not be extracted on another one), but I think those have been abandoned.

I'm not sure if this is necessary, but I can add a check for some reasonable maximum path length, if you prefer. If you want that, what should the limit be? `2^16` seems to be way beyond what I would expect to see for real pathnames, but small enough not to cause memory issues...

Reply:
Hmm, I suppose this is OK then, but I would probably want to move this logic to the `z.extractFile` method, if possible.

Reply:
I put the logic into `extractNext` because it already had done the type assertion on `header`, but you are right, it must be in `extractFile`, otherwise `Extract` will not be able to extract symlinks.  I'll do this (and duplicate the `header` assertion into `extractFile`).

Unfortunately this points out that test coverage of the `Extract` method is lacking. 😦